src/main/java/com/ibm/disni/util/MemoryAllocation.java

```diff
@@ -21,67 +21,155 @@
 
 package com.ibm.disni.util;
 
+import org.slf4j.Logger;
+
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
+import java.util.Comparator;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
 
-public class MemoryAllocation {
-	public enum MemType {
-		INDIRECT, DIRECT,
-	}
 
+public class MemoryAllocation {
+	private static final int MIN_BLOCK_SIZE = 64; // 64B
+	private int MAX_CACHE_SIZE = 10 * 1024 * 1024; // 10MB
```

Comment:
why 10MB? Seems arbitrary

Reply:
Tested on resource hungry spark application with a lot of QPs and a lot of WQ processing. It was about 10 MB of buffers in the cache. But it possible to change this value though API.